src/switchroot/ostree-mount-util.h

```diff
@@ -114,6 +120,46 @@ read_proc_cmdline_key (const char *key)
   return ret;
 }
 
+static inline char *
+get_aboot_root_slot (void)
+{
+  char __attribute__ ((cleanup (free_char))) *slot_suffix
+      = read_proc_cmdline_key ("androidboot.slot_suffix");
+  if (slot_suffix[1] == 'a')
+    return strdup ("/ostree/root.a");
+  else if (slot_suffix[1] == 'b')
+    return strdup ("/ostree/root.b");
+
+  errx (EXIT_FAILURE, "androidboot.slot_suffix invalid: %s", slot_suffix);
+
+  return NULL;
+}
+
+static inline char *
+move (char **to_move)
+{
+  char *ret = *to_move;
+  *to_move = NULL;
+
+  return ret;
+}
+
+static inline char *
+get_ostree_target (void)
+{
+  char __attribute__ ((cleanup (free_char))) *ostree_cmdline = read_proc_cmdline_key ("ostree");
+
+  if (!ostree_cmdline)
+    errx (EXIT_FAILURE, "No ostree= cmdline");
+
+  if (strcmp (ostree_cmdline, "aboot"))
```

Comment:
Personally i like strcmp calls to explicitly compare with zero. so `== 0` or ` != 0` just because the result really isn't a traditional boolean.